actors/builtin/paych/paych_test.go

```diff
@@ -0,0 +1,179 @@
+package paych_test
+
+import (
+	"context"
+	"testing"
+
+	addr "github.com/filecoin-project/go-address"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	. "github.com/filecoin-project/specs-actors/actors/builtin/paych"
+	"github.com/filecoin-project/specs-actors/actors/crypto"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestPaymentChannelActor_Constructor(t *testing.T) {
+	ctx := context.Background()
+	actor := pcActorHarness{Actor{}, t}
+
+	pcaAddr := tutil.NewIDAddr(t, 100)
+	callerAddr := tutil.NewIDAddr(t, 101)
+
+	t.Run("can create a payment channel actor", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, pcaAddr).
+			WithCaller(callerAddr, builtin.AccountActorCodeID).
+			WithReceiverType(builtin.AccountActorCodeID)
+
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt, pcaAddr, callerAddr)
+	})
+
+	t.Run("fails if caller is not account actor", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, pcaAddr).WithCaller(callerAddr, builtin.CronActorCodeID)
+		rt := builder.Build(t)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID)
+		rt.ExpectAbort(18, func() {
+			rt.Call(actor.Constructor, &ConstructorParams{To: pcaAddr})
+		})
+	})
+
+	t.Run("fails if target is not account actor", func(t *testing.T) {
+		builder := mock.NewBuilder(ctx, pcaAddr).
+			WithCaller(callerAddr, builtin.AccountActorCodeID).
+			WithReceiverType(builtin.CronActorCodeID)
+		rt := builder.Build(t)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID)
+		expMsg := "target actor [0 - 64] must be an account (bafkqadlgnfwc6mjpmfrwg33vnz2a), was bafkqactgnfwc6mjpmnzg63q"
```

Comment:
Please don't add assertions about abort messages. I want maximum freedom for us to improve these strictly-diagnostic messages later. It's sufficient to check the exit code, even if that doesn't currently uniquely identify the cause.